pdfly/up2.py

```diff
@@ -29,3 +29,45 @@ def main(pdf: Path, output: Path) -> None:
     with open(output, "wb") as fp:
         writer.write(fp)
     print("done.")
+
+def up2_main(pdf: Path, output: Path, layout: str = None) -> None:
+    reader = PdfReader(str(pdf))
+    writer = PdfWriter()
+
+    if layout:
+        # Define layout configurations (columns, rows) for each grid type
+        layout_options = {
+            "2x2": (2, 2),
+            "3x3": (3, 3),
+            "1x2": (1, 2),
+            "2x1": (2, 1)
+        }
```

Comment:
What is the reason to restrict the layout options? Wouldn't any XxY with X and Y being positive integers work?